qcodes/station.py

```diff
@@ -179,3 +217,202 @@ def __getitem__(self, key):
         return self.components[key]
 
     delegate_attr_dicts = ['components']
+
+
+    def load_config_file(self, filename: Optional[str] = None):
+
+        # 1. load config from file
+        if use_pyyaml:
+            import yaml
+        else:
+            yaml = YAML()
+        if filename is None:
+            filename = DEFAULT_CONFIG_FILE
+        try:
+            with open(filename, 'r') as f:
+                self._config = yaml.load(f)
+        except FileNotFoundError as e:
+            try:
+                if not os.path.isabs(filename) and DEFAULT_CONFIG_FOLDER is not None:
+                    with open(os.path.join(DEFAULT_CONFIG_FOLDER, filename),
+                              'r') as f:
+                        self._config = yaml.load(f)
+                else:
+                    raise e
+            except FileNotFoundError:
+                log.warning('Could not load default config for Station: ' + 
+                            e.msg)
+                return
+                    
+
+        self._instrument_config = self._config['instruments']
+
+        # 2. add config to snapshot componenents
+        class ConfigComponent:
+            def __init__(self, data):
+                self.data = data
+
+            def snapshot(self, update=True):
+                return self.data
+
+        # this overwrites any previous configuration
+        # but does invoke snapshoting
+        self.components['StationConfigurator'] = ConfigComponent(self._config)
+
+
+        # 3. create shortcut methods to instantiate instruments via
+        # `load_<instrument_name>()` so that autocompletion can be used
+        # first remove methods that have been added by a previous
+        # `load_config_file` call
+        while len(self._added_methods):
+            delattr(self, self._added_methods.pop())
+
+        # add shortcut methods
+        for instrument_name in self._instrument_config.keys():
+            # TODO: check if name is valid (does not start with digit, contain
+            # dot, other signs etc.)
+            method_name = f'load_{instrument_name}'
+            if method_name.isidentifier():
+                setattr(self, method_name, partial(
+                    self.load_instrument, identifier=instrument_name))
+
+                self._added_methods.append(method_name)
+            else:
+                log.warning(f'Invalid identifier: ' +
+                            f'for the instrument {instrument_name} no ' +
+                            f'lazy loading method {method_name} could be ' +
+                            'created in the StationConfigurator')
+
+
+    def load_instrument(self, identifier: str,
```

Comment:
Some of the functionality of this function is already implemented in [qcodes/instrument/base.py -> find_or_create_instrument](https://github.com/QCoDeS/Qcodes/blob/4a737aa3a7001d5adbff026c107f3d704ebfdea5/qcodes/instrument/base.py#L730). Could you have a look at it and re-use/merge/else, if it makes sense?

Reply:
I had exactly that in mind. This is why I added the `revive_instance` kwarg. The reason I am not calling the 'find_or_create_instrument` function is that I need to do stuff in the different cases... (like removing parameters from the monitor) I think I will do a pr on the monitor and another one to refactor these two occurrences of similiar code...